### PR TITLE
Fixes #117 - Warn if body/head insertion comments are missing

### LINKED ISSUE #117: Warn if base HTML payload is missing BODY and/or HEAD insertion comments
A non-fastbooted payload is served if the insertion comments are missing from the base HTML. For those using an older version of [ember-cli-html-minifier](https://github.com/gdub22/ember-cli-html-minifier), these comments (and all comments!) are stripped and fastboot fails silently.

The aforementioned addon has added in [fastboot compatibility](https://github.com/gdub22/ember-cli-html-minifier/pull/5), but I think a warning on the fastboot side is prudent as well.

Here's the offending fastboot code: https://github.com/ember-fastboot/fastboot/blob/master/src/result.js#L116-L122

### PR DESCRIPTION
**Pull request by briangonzalez**
Fixes #117

### CODE REVIEW

**Comment by briangonzalez** [comment]
@kratiahuja @danmcclain @arjansingh

File: src/result.js [hunk 1]
```diff
@@ -113,10 +115,20 @@ class Result {
 }
 
 function insertIntoIndexHTML(html, head, body) {
-  html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", body);
+  const headComment = "<!-- EMBER_CLI_FASTBOOT_HEAD -->";
+  const bodyComment = "<!-- EMBER_CLI_FASTBOOT_BODY -->";
+
+  const warnMissing = function (missing) {
```
Comment (kratiahuja) [inline src/result.js]: Can you use `() => {..}` instead now that we dropped Node 0.12.

File: src/result.js [hunk 2]
```diff
@@ -113,10 +115,20 @@ class Result {
...
+    chalk.red(`Fastboot was not able to find ${missing} in base HTML`);
```
Comment (kratiahuja) [inline src/result.js]: I think we should throw an error as well. We don't expect developers to use `fastboot` library and call `result.html()` without including `ember-cli-fastboot` addon. They go hand in glove.

**Comment by briangonzalez** [comment]
@kratiahuja There ya go 😉

File: src/result.js [hunk 3]
```diff
@@ -113,10 +115,20 @@ class Result {
 }
 
 function insertIntoIndexHTML(html, head, body) {
-  html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", body);
+  const headComment = "<!-- EMBER_CLI_FASTBOOT_HEAD -->";
+  const bodyComment = "<!-- EMBER_CLI_FASTBOOT_BODY -->";
+
+  const warnMissing = missing => {
```
Comment (kratiahuja) [inline src/result.js]: Rename this function to something better since you aren't warning now but throwing an error.

Comment (rwjblue) [inline src/result.js]: agreed

**Comment by kratiahuja** [comment]
Closing this since there is no activity and #149 supersedes this.

**Pull request closed without merging**